src/main/java/dwbh/api/services/internal/checkers/NotNull.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) 2019 Kaleidos Open Source SL
+ *
+ * This file is part of Don't Worry Be Happy (DWBH).
+ * DWBH is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * DWBH is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with DWBH.  If not, see <https://www.gnu.org/licenses/>
+ */
+package dwbh.api.services.internal.checkers;
+
+import static dwbh.api.util.Check.checkIsTrue;
+
+import dwbh.api.util.Check;
+import dwbh.api.util.ErrorConstants;
+import dwbh.api.util.Result;
+
+/**
+ * This checker expects the argument passed to be not null, otherwise it will return a failing
+ * {@link Result}
+ *
+ * @since 0.1.0
+ */
+public class NotNull {
```

Comment:
Mmmm I don't know. I think is because we're not used to it.  

For me **the main reason** is that the suffix `Check` or `Checker` make them too long (specially if then the instance names should follow the naming).

```java
UserIsInGroup userIsInGroup = new UserIsInGroup(userGroupRepository);
UserIsInGroupCheck userIsInGroupCheck = new UserIsInGroupCheck(userGroupRepository);
```

Another ideas that could help justifying the decision:

- They're in the `dwbh.api.checkers` package, so the suffix may seem redundant

```java
import dwbh.api.services.internal.checkers.NotNullChecker;
```

- They're business rules, with a method to check that the business rule is met. That method is coincidentally `check()`, but could have been `validate()` as well.
- There're  [java annotation validations](https://docs.oracle.com/javaee/7/api/javax/validation/constraints/package-summary.html) , as the `@NotNull` annotation, and nobody misses a suffix there.

```java
void doSomething(@NotNull String name {
/* ... */
}
void doSomething(@NotNullChecker String name) {
/* ... */
}

Reply:
Still don't know, one more example. I'm not sure adding the `Check` suffix helps reading the code better:

```java
 public Result<Vote> createVote(CreateVoteInput input) {
    Group group = votingRepository.findGroupByUserAndVoting(input.getUserId(), input.getVotingId());

    VoteScoreBoundaries voteScoreBoundaries = new VoteScoreBoundaries();
    UserOnlyVotedOnce userOnlyVotedOnce = new UserOnlyVotedOnce(votingRepository);
    VotingHasExpired votingHasExpired = new VotingHasExpired(votingRepository);
    NotNull notNull = new NotNull();
    VoteAnonymousAllowedInGroup anonymousAllowed = new VoteAnonymousAllowedInGroup();

    return Result.<Vote>create()
        .thenCheck(() -> voteScoreBoundaries.check(input.getScore()))
        .thenCheck(() -> userOnlyVotedOnce.check(input))
        .thenCheck(() -> votingHasExpired.check(input.getVotingId()))
        .thenCheck(() -> notNull.check(group))
        .thenCheck(() -> anonymousAllowed.check(input.isAnonymous(), group.isAnonymousVote()))
        .then(() -> createVoteIfSuccess(input));
  }
```

vs 

```java
public Result<Vote> createVote(CreateVoteInput input) {
    Group group = votingRepository.findGroupByUserAndVoting(input.getUserId(), input.getVotingId());

    VoteScoreBoundariesCheck scoreBoundariesCheck = new VoteScoreBoundariesCheck();
    UserOnlyVotedOnceCheck userOnlyVotedOnceCheck = new UserOnlyVotedOnceCheck(votingRepository);
    VotingHasExpiredCheck votingHasExpiredCheck = new VotingHasExpiredCheck(votingRepository);
    NotNullCheck notNullCheck = new NotNullCheck();
    VoteAnonymousAllowedInGroupCheck allowedInGroupCheck = new VoteAnonymousAllowedInGroupCheck();

    return Result.<Vote>create()
        .thenCheck(() -> scoreBoundariesCheck.check(input.getScore()))
        .thenCheck(() -> userOnlyVotedOnceCheck.check(input))
        .thenCheck(() -> votingHasExpiredCheck.check(input.getVotingId()))
        .thenCheck(() -> notNullCheck.check(group))
        .thenCheck(() -> allowedInGroupCheck.check(input.isAnonymous(), group.isAnonymousVote()))
        .then(() -> createVoteIfSuccess(input));
  }
```
Anyway, if after this you still wants to change them to `XXXCheck` no problem, I'm changing them.

Reply:
Or maybe this alternative to know where to find all checkers:

```java
public Result<Vote> createVote(CreateVoteInput input) {
    Group group = votingRepository.findGroupByUserAndVoting(input.getUserId(), input.getVotingId());

    VoteScoreBoundariesCheck scoreBoundariesCheck = Checkers.voteScoreBoundaries(votingRepository);
    UserOnlyVotedOnceCheck userOnlyVotedOnceCheck = Checkers.userOnlyVotedOnceCheck(votingRepository);
    NotNullCheck notNullCheck = Checkers.notNull();
    VoteAnonymousAllowedInGroupCheck allowedInGroupCheck = Checkers. voteAnonymousAllowedInGroupCheck();

    return Result.<Vote>create()
        .thenCheck(() -> scoreBoundariesCheck.check(input.getScore()))
        .thenCheck(() -> userOnlyVotedOnceCheck.check(input))
        .thenCheck(() -> votingHasExpiredCheck.check(input.getVotingId()))
        .thenCheck(() -> notNullCheck.check(group))
        .thenCheck(() -> allowedInGroupCheck.check(input.isAnonymous(), group.isAnonymousVote()))
        .then(() -> createVoteIfSuccess(input));
  }
```

with the jdk 10 variant:

```java
public Result<Vote> createVote(CreateVoteInput input) {
    Group group = votingRepository.findGroupByUserAndVoting(input.getUserId(), input.getVotingId());

    var scoreBoundariesCheck = Checkers.voteScoreBoundaries(votingRepository);
    var userOnlyVotedOnceCheck = Checkers.userOnlyVotedOnceCheck(votingRepository);
    var notNullCheck = Checkers.notNull();
    var allowedInGroupCheck = Checkers.voteAnonymousAllowedInGroupCheck();

    return Result.<Vote>create()
        .thenCheck(() -> scoreBoundariesCheck.check(input.getScore()))
        .thenCheck(() -> userOnlyVotedOnceCheck.check(input))
        .thenCheck(() -> votingHasExpiredCheck.check(input.getVotingId()))
        .thenCheck(() -> notNullCheck.check(group))
        .thenCheck(() -> allowedInGroupCheck.check(input.isAnonymous(), group.isAnonymousVote()))
        .then(() -> createVoteIfSuccess(input));
  }
```

Reply:
I see your points. I'm not fully convinced, but I really think that the issue doesn't merit the time to change it, so I'll agree with leave it as is.